src/easyvk.cpp

```diff
@@ -627,64 +650,66 @@ namespace easyvk
     specMapContent[1] = 1;
     specMap[2] = VkSpecializationMapEntry{2, 8, sizeof(uint32_t)};
     specMapContent[2] = 1;
-
     // key is index, value is length
-    for (const auto &[key, value] : workgroupMemoryLengths)
-    {
+    for (const auto &[key, value] : workgroupMemoryLengths) {
       specMap[3 + key] = VkSpecializationMapEntry{3 + key, (3 + key) * 4, sizeof(uint32_t)};
       specMapContent[3 + key] = value;
     }
 
     VkSpecializationInfo specInfo{numSpecConstants, specMap, numSpecConstants * sizeof(uint32_t), specMapContent};
 
     // Define shader stage create info
-    VkPipelineShaderStageCreateInfo stageCI{
-        VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO,
-        nullptr,
-        VK_PIPELINE_SHADER_STAGE_CREATE_REQUIRE_FULL_SUBGROUPS_BIT,
```

Comment:
I'm not sure what the right option to optionally enable it here is - we could of course make it an optional flag on `Program::initialize()` but I'm wary of complicating that interface much more, especially for something most work isn't going to touch. Maybe a preprocessor directive? (i.e., `#DEFINE EASYVK_FULL_SUBGROUPS`) Seems a lot easier than having to introduce either a set of flags, bit vector, or something else to enable features in easyvk we're likely to use less often

Reply:
Every option is going to be "dirty" in some sense. Vulkan itself handles this via giant structs with all the optional features that you can enable, which is kinda what the `BufferInfo` struct was doing.

Generally, I would say I'm opposed to runtime flags like preprocessor directives to enable certain features, my opinion is that it is more brittle and confusing then writing it into application code.

Reply:
That's true, preprocessor directives aren't especially clean either. I'm content just adding an optional `VkPipelineShaderStageCreateFlags` to `Program::initialize()`, since it should just let you add an arg to the Program in any code needing it, but yeah, we're definitely toeing a similar unclear line to what Vulkan does to expose this stuff - as long as easyvk continues to make writing Vulkan easier!